zio-http/shared/src/main/scala/zio/http/codec/HttpCodec.scala

```diff
@@ -2264,140 +2269,220 @@ object HttpCodec extends ContentCodecs with HeaderCodecs with MethodCodecs with
+        case _: Schema.Optional[_]                                                     => true
+        case s: Schema.Transform[_, _, _] if s.schema.isInstanceOf[Schema.Optional[_]] => true
+        case _                                                                         => false
       }
 
-      private def unlazy(field: Schema.Field[Any, Any]): Schema.Field[Any, Any] = field.schema match {
-        case Schema.Lazy(schema) =>
-          Schema.Field(
-            field.name,
-            schema(),
-            field.annotations,
-            field.validation,
-            field.get,
-            field.set,
+    val isPrimitive: Boolean = schema match {
+      case _: Schema.Primitive[_]                                                     => true
+      case s: Schema.Optional[_] if s.schema.isInstanceOf[Schema.Primitive[_]]        => true
+      case s: Schema.Transform[_, _, _] if s.schema.isInstanceOf[Schema.Primitive[_]] => true
+      case _                                                                          => false
+    }
+
+    val isRecord: Boolean = schema match {
+      case _: Schema.Record[_]                                                     => true
+      case s: Schema.Optional[_] if s.schema.isInstanceOf[Schema.Record[_]]        => true
+      case s: Schema.Transform[_, _, _] if s.schema.isInstanceOf[Schema.Record[_]] => true
+      case _                                                                       => false
+    }
+
+    def optional: SchemaCodec[Option[A]] = copy(schema = schema.optional)
+
+    val recordFields: Chunk[(Schema.Field[_, _], SchemaCodec[Any])] = {
+      val fields = schema match {
+        case record: Schema.Record[A]                                                =>
+          record.fields
+        case s: Schema.Optional[_] if s.schema.isInstanceOf[Schema.Record[_]]        =>
+          s.schema.asInstanceOf[Schema.Record[A]].fields
+        case s: Schema.Transform[_, _, _] if s.schema.isInstanceOf[Schema.Record[_]] =>
+          s.schema.asInstanceOf[Schema.Record[A]].fields
+        case _                                                                       => Chunk.empty
+      }
+      fields.map(unlazyField).map {
+        case field if field.schema.isInstanceOf[Schema.Collection[_, _]] =>
+          val elementSchema = field.schema.asInstanceOf[Schema.Collection[_, _]] match {
+            case s: Schema.NonEmptySequence[_, _, _] => s.elementSchema
+            case s: Schema.Sequence[_, _, _]         => s.elementSchema
+            case s: Schema.Set[_]                    => s.elementSchema
+            case _: Schema.Map[_, _]                 => throw new IllegalArgumentException("Maps are not supported")
+            case _: Schema.NonEmptyMap[_, _]         => throw new IllegalArgumentException("Maps are not supported")
+          }
+          val codec         = SchemaCodec(Some(if (!kebabCase) field.name else camelToKebab(field.name)), elementSchema)
```

Comment:
The capitalization is not important? e.g. `Max-Age` versus `maX-aGE`?

Reply:
RFC says header names are case-insensitive